crates/bevy_ecs/src/archetype.rs

```diff
@@ -372,7 +375,7 @@ impl Default for Archetypes {
         let mut archetypes = Archetypes {
             archetypes: Vec::new(),
             archetype_ids: Default::default(),
-            archetype_component_count: 0,
+            archetype_component_count: 1, // 0 is reserved
```

Comment:
This should note what it's reserved for.

Reply:
Done.